### PR TITLE
refactor: image url resolver

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

### CODE REVIEW

File: packages/web3-shared/base/src/utils/fetch.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/**
+ * Fetch image by creating an image element.
+ * It's not supporting in mv3 on background page.
+ * @param url
+ * @returns
+ */
+export function fetchImageByDOM(url: string) {
```
Comment (UncleBill) [inline packages/web3-shared/base/src/utils/fetch.ts]:
```suggestion
export function fetchImageViaDOM(url: string) {
```

File: packages/web3-shared/base/src/utils/fetch.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    })
+}
+
+/**
+ * Fetch image by fetching with HTTP request.
+ * This manner will constraint by browser safe policy (CORS).
+ * @param url
+ * @returns
+ */
+export async function fetchImageByHTTP(url: string) {
```
Comment (UncleBill) [inline packages/web3-shared/base/src/utils/fetch.ts]:
```suggestion
export async function fetchImageViaHTTP(url: string) {
```

File: packages/web3-shared/base/src/utils/fetch.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    return new Promise<Blob | null>((resolve, reject) => {
+        const img = document.createElement('img')
+        const cleanup = () => {
+            img.removeEventListener('load', onload)
+            img.removeEventListener('error', reject)
+        }
+        const onload = () => {
+            const canvas = document.createElement('canvas')
+            const context = canvas.getContext('2d')
+            canvas.height = img.naturalHeight
```
Comment (UncleBill) [inline packages/web3-shared/base/src/utils/fetch.ts]: Use https://developer.mozilla.org/zh-CN/docs/Web/API/OffscreenCanvas instead.

Comment (guanbinrui) [reply packages/web3-shared/base/src/utils/fetch.ts]: It's an experiment API. I'm hesitant to use it.

**Pull request merged**